src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -2429,4 +2442,345 @@ public Response getTimestamps(@PathParam("identifier") String id) {
+                            }
+                        } else {
+                            return error(BAD_REQUEST,
+                                    "You must upload a file or provide a storageidentifier, filename, and mimetype.");
+                        }
+
+                        msg("ADD!  = " + newFilename);
+
+                        //-------------------
+                        // Run "runAddFileByDatasetId"
+                        //-------------------
+
+                        addFileHelper.runAddFileByDataset(dataset,
+                                newFilename,
+                                newFileContentType,
+                                newStorageIdentifier,
+                                null,
+                                optionalFileParams, true);
+
+                        if (addFileHelper.hasError()) {
+
+                            JsonObjectBuilder fileoutput = Json.createObjectBuilder()
+                                    .add("storageIdentifier ", newStorageIdentifier)
+                                    .add("error Code: ", addFileHelper.getHttpErrorCode().toString())
+                                    .add("message ", addFileHelper.getErrorMessagesAsString("\n"));
+
+                            jarr.add(fileoutput);
+
+                        } else {
+                            String successMsg = BundleUtil.getStringFromBundle("file.addreplace.success.add");
+
+                            JsonObject successresult = addFileHelper.getSuccessResultAsJsonObjectBuilder().build();
+
+                            try {
+                                logger.fine("successMsg: " + successMsg);
+                                String duplicateWarning = addFileHelper.getDuplicateFileWarning();
+                                if (duplicateWarning != null && !duplicateWarning.isEmpty()) {
+                                    // return ok(addFileHelper.getDuplicateFileWarning(), addFileHelper.getSuccessResultAsJsonObjectBuilder());
+                                    JsonObjectBuilder fileoutput = Json.createObjectBuilder()
+                                            .add("storageIdentifier ", newStorageIdentifier)
+                                            .add("warning message: ", addFileHelper.getDuplicateFileWarning())
+                                            .add("message ", successresult.getJsonArray("files").getJsonObject(0));
+                                    jarr.add(fileoutput);
+
+                                } else {
+                                    JsonObjectBuilder fileoutput = Json.createObjectBuilder()
+                                            .add("storageIdentifier ", newStorageIdentifier)
+                                            .add("message ", successresult.getJsonArray("files").getJsonObject(0));
+                                    jarr.add(fileoutput);
+                                }
+
+                            } catch (Exception ex) {
```

Comment:
I'm not sure what is being caught here (how many different types of Exceptions?) but I think it would be useful to differentiate between them.